server/instance.go

```diff
@@ -71,6 +82,39 @@ func (s *Server) registerInstanceRoutes(g *echo.Group) {
 		return nil
 	})
 
+	g.POST("/instance/new-embedded-pg", func(c echo.Context) error {
+		pgUser := "postgres"
+		commonPreffix := "bb-pg"
+		randomSuffix := string(rune(time.Now().Unix()))
+
+		baseDir := fmt.Sprintf("%s/%s_%s", common.GetPostgresSocketDir(), commonPreffix, randomSuffix)
+		resourceDir := path.Join(baseDir, "resources")
+		dataDir := path.Join(baseDir, "pgdata")
+
+		pgInstance, err := postgres.Install(resourceDir, dataDir, pgUser)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to install embedded postgres instance").SetInternal(err)
+		}
+
+		port, err := common.GetFreePort()
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to get free port").SetInternal(err)
+		}
+
+		err = pgInstance.Start(port, os.Stderr, os.Stderr)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, "Failed to start embedded postgres instance").SetInternal(err)
+		}
+
+		connectionInfo := pgConnectionInfo{
+			Host:     common.GetPostgresSocketDir(),
+			Port:     port,
```

Comment:
We can use a fixed port here instead of finding an unused port. PostgreSQL server isn't actually listening to the port but using this port as Unix domain socket file name.

Reply:
I think we can allow users to start many embedded pg instances instead of just one.

Reply:
Use `23333` as a fixed value for port and `tmp-pgdata` for dataDir.